src/core/MpiCallbacks.hpp

```diff
@@ -544,9 +544,9 @@ class MpiCallbacks {
    *
    * This method can only be called on the head node.
    */
-  template <class R, class... Args>
+  template <class R, class... Args, class... ArgRef>
```

Comment:
do you think that we should make sure that ```Args``` and ```ArgRef``` are mutually convertible via SFINAE?

Reply:
There is no other upload that could be picked up, so actually substitution failure is an error here.